src/IO/ReadBufferFromS3.cpp

```diff
@@ -323,16 +323,23 @@ std::unique_ptr<ReadBuffer> ReadBufferFromS3::initialize()
     if (read_settings.for_object_storage)
         ProfileEvents::increment(ProfileEvents::DiskS3GetObject);
 
+    // We do not know in advance how many bytes we are going to consume, to avoid blocking estimated it from below
```

Comment:
Actually we can know it pretty accurately (with some small impresicion if the case the query does not have `LIMIT` clause), need to pass it from the underlying buffer which has this info. Does it make sense to use this information? But in case there is `LIMIT` this information might be totally incorrect and we cannot know that on this level. But in other cases it will be pretty accurate.

Reply:
This inaccuracy should not be a big deal, let's just not bother too much. In a long run with adjust() calls right amount of bytes should be accounted.